esp/esdllib/esdl_script.cpp

```diff
@@ -2310,6 +2340,327 @@ class CEsdlTransformOperationFunction : public CEsdlTransformOperationWithChildr
     }
 };
 
+/**
+ * @brief Helper structure used by both tx-summary and tx-summary-timer.
+ *
+ * There is common data used by both operations. Because one operation is childless and the other
+ * allows children, they cannot have a common base class. Instead each contains an instance of this
+ * structure to provide common handling of the common data.
+ */
+struct TxSummaryCoreInfo
+{
+    StringAttr m_name;
+    unsigned   m_level = LogMin;
+    unsigned   m_groups = TXSUMMARY_GRP_ENTERPRISE;
+
+    TxSummaryCoreInfo(StartTag& stag, const char* tagname, const char* traceName, bool ignoreCodingErrors)
+    {
+        m_name.set(stag.getValue("name"));
+        if (m_name.isEmpty())
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, tagname, "without name", traceName, !ignoreCodingErrors);
+        const char* level = stag.getValue("level");
+        if (!isEmptyString(level))
+        {
+            if (strieq(level, "min"))
+                m_level = LogMin;
+            else if (strieq(level, "normal"))
+                m_level = LogNormal;
+            else if (strieq(level, "max"))
+                m_level = LogMax;
+            else if (TokenDeserializer().deserialize(level, m_level) != Deserialization_SUCCESS || m_level < LogMin || m_level > LogMax)
+                esdlOperationError(ESDL_SCRIPT_InvalidOperationAttr, tagname, "invalid level", traceName, !ignoreCodingErrors);
+        }
+        const char* coreGroup = stag.getValue("core_group");
+        if (!isEmptyString(coreGroup) && strToBool(stag.getValue("core_group")))
+            m_groups |= TXSUMMARY_GRP_CORE;
+    }
+};
+
+static TokenDeserializer s_deserializer;
+
+class CEsdlTransformOperationTxSummary : public CEsdlTransformOperationWithoutChildren
+{
+protected:
+    enum Type { Text, Signed, Unsigned, Decimal };
+    enum Mode { Append, Set };
+    TxSummaryCoreInfo     m_info;
+    Owned<ICompiledXpath> m_xValue;
```

Comment:
Yeah I find that select/value dichotomy very confusing. 2 is my least favorite option. 1 would probably be my first choice. 3 is OK but I'm not sure about starting a new precedent that likely won't be retrofitted to previous operations. What actually is the benefit to having separate attributes for literal & xpath? Do we just avoid the overhead of compiling the xpath?

This may be a case where Tony's input about how he sees the direction has most weight.

Reply:
> What actually is the benefit to having separate attributes for literal & xpath?

It does avoid compiling and evaluating an XPath every time through a script, but it also allows scripts to use `"some text here"` instead of `"'some text here'"`. It looks cleaner, in my opinion, but that is subjective.

Reply:
After discussion with Tony, `value` will be removed (leaving only `select`). Should something like #3 be desirable in the future, changes will be applied to all relevant operations.